packages/devextreme/js/__internal/ui/html_editor/ui/m_aiDialog.ts

```diff
@@ -0,0 +1,207 @@
+import localizationMessage from '@js/common/core/localization/message';
+import $ from '@js/core/renderer';
+import { extend } from '@js/core/utils/extend';
+import type { ToolbarItem } from '@js/ui/popup';
+import SelectBox from '@js/ui/select_box';
+import TextArea from '@js/ui/text_area';
+
+import type { CommandDefinition, CommandsMap } from '../utils/m_ai';
+import DialogBase from './m_baseDialog';
+
+const AI_DIALOG_COMMANDS_WITH_OPTIONS = ['translate', 'changeStyle', 'changeTone'];
+
+export interface AIDialogShowPayload {
+  currentCommand: string;
+  currentOption?: string;
+  text?: string;
+  commandsMap: CommandsMap;
+}
+
+const AI_DIALOG_CONTROLS_CLASS = 'dx-aidialog-controls';
+const AI_DIALOG_CONTENT_CLASS = 'dx-aidialog-content';
+
+export default class AIDialog extends DialogBase {
+  private _loading = false;
+
+  private readonly _serviceAI;
+
+  private _commandsMap: CommandsMap = {};
+
+  private _currentCommand?: string;
+
+  private _currentOption?: string;
+
+  private _optionsList?: string[];
+
+  private _resultText = '';
+
+  private _commandBoxInstance?: SelectBox;
+
+  private _optionBoxInstance?: SelectBox;
+
+  private _textAreaInstance?: TextArea;
```

Comment:
[naming]
I think it's too abstract now
```suggestion
  private _resultTextArea?: TextArea;
```

I think you can also omit `instance` in another names if you change "Box" with "SelectBox"